langchain/src/vectorstores/tests/typesense.test.ts

```diff
@@ -0,0 +1,112 @@
+import { Client } from "typesense";
+import { Document } from "../../document.js";
+import { FakeEmbeddings } from "../../embeddings/fake.js";
+import { Typesense } from "../typesense.js";
+
+test("documentsToTypesenseRecords should return the correct typesense records", async () => {
+  const embeddings = new FakeEmbeddings();
+  const vectorstore = new Typesense(embeddings, {
+    schemaName: "test",
+    typesenseClient: {} as unknown as Client,
+    columnNames: {
+      vector: "vec",
+      pageContent: "text",
+      metadataColumnNames: ["foo", "bar", "baz"],
+    },
+  });
+
+  const documents: Document[] = [
+    {
+      metadata: {
+        id: "1",
+        foo: "fooo",
+        bar: "barr",
+        baz: "bazz",
+      },
+      pageContent: "hello world",
+    },
+    {
+      metadata: {
+        id: "2",
+        foo: "foooo",
+        bar: "barrr",
+        baz: "bazzz",
+      },
+      pageContent: "hello world 2",
+    },
+  ];
+
+  const expected = [
+    {
+      text: "hello world",
+      foo: "fooo",
+      bar: "barr",
+      baz: "bazz",
+      vec: await embeddings.embedQuery("hello world"),
+    },
+    {
+      text: "hello world 2",
+      foo: "foooo",
+      bar: "barrr",
+      baz: "bazzz",
+      vec: await embeddings.embedQuery("hello world 2"),
+    },
+  ];
+
+  expect(await vectorstore.documentsToTypesenseRecords(documents)).toEqual(
+    expected
+  );
+});
+
+test("typesenseRecordsToDocuments should return the correct langchain documents", async () => {
```

Comment:
Consider adding integration tests as well

Reply:
I am not sure about how can I do it. I will ask for help in Discord, there was somebody that knows Langchain and maybe can guide me